manimlib/scene/scene.py

```diff
@@ -205,6 +208,9 @@ def add(self, *new_mobjects):
         """
         self.remove(*new_mobjects)
         self.mobjects += new_mobjects
+        for new_mob in new_mobjects:
```

Comment:
One potential issue I see is if after a mobject has been added to a scene, it's internal structure changes (e.g. you add or remove a submobject), then this won't be reflected in scene's list of mob_listeners.

I wonder if it would be better to handle event listening the same way updaters are, where it's mostly taken care of in Mobject rather than Scene.

Reply:
@3b1b ,
That would be great, I'll try to mimic the updaters structure for event listening and create a new pull request.
Just one question, how should I categorize different types of events. Enum or something else? Any suggestions?

Reply:
> @3b1b ,
> That would be great, I'll try to mimic the updaters structure for event listening and create a new pull request.
> Just one question, how should I categorize different types of events. Enum or something else? Any suggestions?

By different types of events, I mean on_mouse_drag, on_key_press, etc.

Reply:
Or I could create an array of each type of event.

Reply:
An array of each type is probably the right way to go.

Reply:
I was thinking of creating an EventHandler class with a type attribute(Enum) and a function attribute. And Mobject will hold an array of EventHandler objects.

Reply:
Then we could add methods for each event type inside Mobject to add a specific event handler.